### DISCUSSION TITLE
Merchant Initiated Transaction

### DISCUSSION THREAD

**Discussion by Mat40100** [PayGreen/documentation#30] [Q&A] [answered]
Bonjour, 

Je souhaite intégrer un système de paiement en plusieurs fois, et pour cela je dois être capable d'initier des Merchant Initiated Transaction.

Lorsque j'envoie ma requête avec merchant_initiated=true, je reçoit des 422, il y'a d'autres params à ajouter ? ou qui deviennent obligatoires?

Je n'ai rien trouvé à ce sujet dans la documentation.

bonne fin de journée

**Comment by AdrienBr** [comment, marked as answer]
Bonjour @Mat40100 ,
Les erreurs 422 sont généralement accompagnées d'un message explicatif dans la réponse API.

La principale règle d'un paiement merchant_initiated est que vous devez indiquer un instrument existant à utiliser pour le paiement.

Bonne journée,
Adrien

**Reply by Mat40100** [reply]
@AdrienBr Merci pour la réponse, effectivement j'ai cette réponse : "The given Instrument has not been authorized by the Buyer" 

Il y a un endpoint pour l'authorisation du moyen de paiement ?

Je vous explique le flow actuel: 

1) il sélectionne le mode de paiement plus plusieurs fois 

2) Via Pgjs react il rempli le formulaire de carte et avec les evenement j'empeche l'avancer vers la suite s'il ne coche pas l'authorisation d'enregistrement des infos de paiement

3) Je souhaite déclancher un paiement MIT depuis le back, et une fois le paiement valider je redirige le client vers la page de validation.

Je précise que je suis dans la sandbox

**Reply by gaelDolle** [reply]
Bonjour,
Vous pouvez initialiser pgjs en mode="instrument", afin de créer un instrument autorisé. Puis utiliser celui-ci pour réaliser le paiement.

```
 paygreenjs.init({
     mode: 'instrument',
      buyer,
      modeOptions: {
        authorizedInstrument: true,
      },
      publicKey,
      paymentMethod: paymentMethod || 'bank_card',
      style,
      lang: 'fr',
    });

```

https://developers.paygreen.fr/docs/paygreenjs-installation
